### DISCUSSION TITLE
Language adaptation

### DISCUSSION THREAD

**Discussion by cm0s** [OpenIDC/mod_auth_openidc#1036] [Q&A] [answered]
Hi there,

Is there any way to translate error messages sent by mod_auth_openidc?

I can use the OIDCHTMLErrorTemplate to set a custom error page, but messages passed as "%s" variable are in English.

I could do a translation using Javascript inside the template HTML file, but it doesn't seem the cleanest way to do it. Being able to do it on the server side will be better.

Thanks for your help.

**Comment by zandbelt** [comment, marked as answer]
using `OIDCHTMLErrorTemplate` and possibly `ErrorDocument` are the only ways of doing that; you can have the `OIDCHTMLErrorTemplate` post to a server-side processing page that translates, it doesn't need to be Javascript/client based

**Reply by cm0s** [reply]
Thank you for your prompt answer !

**Reply by zandbelt** [reply]
FWIW: 2.4.14 will rely on standard Apache ErrorDocument handling which gives you a better chance at translation as well: https://github.com/OpenIDC/mod_auth_openidc/blob/996cc085bad8329a4d44d2eaf4c9cd1c6e2a1f42/ChangeLog#L1-L4
